### PR TITLE
issue-83-detail-profile-user

### PR DESCRIPTION
**Pull request by 0henri0**
Hello, and thanks for contributing to FS-Focus!

## tldr

There are three main goals in this document, depending on the nature of your pr:

- [description](#description): please tell us about your pr
- [checklist](#checklist): please review the checklist that is most closly related to your pr
- [contributors list](#packagejson-contributors): you're one of us now, please add yourself to `package.json` and let the world know!

The following sections provide more detail on each.

**Improve this document**

Please don't hesitate to [ask questions][issues] for clarification, or to [make suggestions][issues] (or a pull request) to improve this document.

## Description

To help the project's maintainers and community to quickly understand the nature of your pull requeset, please create a description that incorporates the following elements:

- [] what is accomplished by the pr
- [] if there is something potentially controversial in your pr, please take a moment to tell us about your choices

## Checklist

Please use the checklist that is most closely related to your pr _(you only need to use one checklist, and you can skip items that aren't applicable or don't make sense)_:

- [fixing typos]()
- [documentation]()
- [bug fix]()
- [new feature]()
- [other]()

### Fixing typos

- [ ] Please review the [readme advice]() section before submitting changes
- [ ] Add your to the [contributors](#packagejson-contributors) array in package.json!

### Documentation

- [ ] Please review the [readme advice](#readme-advice) section before submitting changes
- [ ] Add your info to the [contributors](#packagejson-contributors) array in package.json!

### Bug Fix

- [ ] All existing unit tests are still passing (if applicable)
- [ ] Add new passing unit tests to cover the code introduced by your pr
- [ ] Update the readme (see [readme advice](#readme-advice))
- [ ] Update or add any necessary API documentation
- [ ] Add your info to the [contributors](#packagejson-contributors) array in package.json!

### New Feature

- [ ] If this is a big feature with breaking changes, consider [opening an issue][issues] to discuss first. This is completely up to you, but please keep in mind that your pr might not be accepted.
- [ ] Run unit tests to ensure all existing tests are still passing
- [ ] Add new passing unit tests to cover the code introduced by your pr
- [ ] Update the readme (see [readme advice](#readme-advice))
- [ ] Add your info to the [contributors](#packagejson-contributors) array in package.json!

Thanks for contributing!

## Readme advice

Please review this section if you are updating readme documentation.

**Readme template**

This project uses [verb][] for documentation. Verb generates the project's readme documentation from the [.verb.md](../.verb.md) template in the root of this project.

Make all documentation changes in `.verb.md`, and please _do not edit the readme.md directly_, or your changes might accidentally get overwritten.

**Code comments**

Please add code comments (following the same style as existing comments) to describe any code changes or new code introduced by your pull request.

**Optionally build the readme**

Any changes made `.verb.md` and/or code comments will be automatically incorporated into the README documentation the next time `verb` is run.

We can take care of building the documentation for you when we merge in your changes, or feel free to run [verb][] yourself. Whatever you prefer is fine with us.

## package.json contributors

**Add yourself!**

When adding your information to the `contributors` array in package.json, please use the following format (provide your name at minimum, the other fields are optional):

```
full name <email@address.com> (https://website.com)
```

**Example**

```json
// -- package.json --
{
  "name": "<%= name %>",
  "contributors": [
    "Brian Woodward <jon.schlinkert@sellside.com> (https://github.com/jonschlinkert)",
    "Jon Schlinkert <brian.woodward@sellside.com> (https://github.com/doowb)",
  ]
}
```

_(If a `contributors` array does not already exist, please feel free to add one wherever you want, and congratulations on being the first to contribute!)_

[issues]: ../../issues
[verb]: https://github.com/verbose/verb

### CODE REVIEW

File: app/Article.php [hunk 1]
```diff
@@ -192,4 +192,19 @@ public function setContentAttribute($value)
 
         $this->attributes['content'] = json_encode($data);
     }
+
+    public function upvoteUsers()
+    {
+        return $this->belongsToMany(\App\User::class, 'followables', 'followable_id', 'user_id')
```
Comment (trantrongbinh) [inline app/Article.php]: ko dùng User::class thôi mà phải cần thêm \App\

File: app/Helpers.php [hunk 2]
```diff
@@ -87,6 +87,28 @@ function getVote($post)
...
+if (!function_exists('getVote')) {
+    /**
+     * Get resource vote off a post.
+     *
+     * @param collection $post
+     * @return array
+     */
+    function getVote($post)
+    {
+        $isUpvote = in_array(auth()->user()->id, $post->upvoteUsers->pluck('id')->toArray()) ? true : false;
```
Comment (trantrongbinh) [inline app/Helpers.php]: hình như auth()->id đc sao ấy nhỉ.

File: app/Helpers.php [hunk 3]
```diff
@@ -87,6 +87,28 @@ function getVote($post)
...
+        $isDownvote = in_array(auth()->user()->id, $post->downvoteUsers->pluck('id')->toArray()) ? true : false;
+        $vote = [
+            'is_up_voted' => $isUpvote,
+            'is_down_voted' => $isDownvote,
+            'is_voted' => $isUpvote || $isDownvote,
+            'vote_count' => count($post->upvoteUsers) + count($post->downvoteUsers),
```
Comment (trantrongbinh) [inline app/Helpers.php]: dùng hàm sum có phải đẹp hơn ko! =))

File: app/Repositories/ArticleRepository.php [hunk 4]
```diff
@@ -325,4 +329,70 @@ public function getRelatedPosts($article)
...
     }
+
+    public function articleIndex()
+    {
+        $hotPosts  = $this->model->select('slug', 'title', 'user_id')
+            ->withCount('comments')
+            ->take(5)
+            ->get();
+
+        $public = Category::withCount('articles')->whereNull('user_id')->OrderBy('articles_count', 'desc')->get();
```
Comment (trantrongbinh) [inline app/Repositories/ArticleRepository.php]: tên biến chả hiểu gì, đặt tên dễ hiểu chút.

Comment (trantrongbinh) [inline app/Repositories/ArticleRepository.php]: bõ công sửa thì sửa lại tên biến cho dễ hiểu chút: $publicCategory

File: resources/views/article/show.blade.php [hunk 5]
```diff
@@ -147,7 +147,7 @@
                                 </div>
                                 <div class="text-small text-light__grey truncate-line__1">
                                     <a href="#">
-                                        {{ $article->published_at->diffForHumans() }}
+                                        {{ $article->published_at }}
```
Comment (trantrongbinh) [inline resources/views/article/show.blade.php]: cái này để diffForHumans để định dạng lại ngày giờ.

File: resources/views/modules/hot-post.blade.php [hunk 6]
```diff
@@ -1,4 +1,6 @@
 <div class="blog-posts">
+
```
Comment (trantrongbinh) [inline resources/views/modules/hot-post.blade.php]: thêm dòng làm gì?

File: resources/views/modules/hot-post.blade.php [hunk 7]
```diff
@@ -1,4 +1,6 @@
...
+
     @foreach ($hotPosts as $post)
         <div class="item d-flex align-items-center">
            {{--  <a href="/user/{{ $post->user->name }}">
```
Comment (trantrongbinh) [inline resources/views/modules/hot-post.blade.php]: xóa hẳn đi

File: resources/views/modules/hot-post.blade.php [hunk 8]
```diff
@@ -7,21 +9,24 @@
...
                     <a href="/user/{{ $post->user->name }}" class="author-name">{{ $post->user->name }}</a>
                 </span>
+
                 <div class="d-flex align-items-center">
                     <div class="views"><i class="far fa-eye"></i> {{ $post->getViews() }}</div>
                     <span class="dot">&middot;</span>
                     <div class="comments"><i class="far fa-comment-alt"></i> {{ $post->comments_count }}</div>
                     <span class="dot">&middot;</span>
-                    <div class="time last--icon"><i class="far fa-clock"></i> {{ $post->published_at->diffForHumans() }}</div>
+                    <div class="time last--icon"><i class="far fa-clock"></i> {{ $post->published_at }}</div>
```
Comment (trantrongbinh) [inline resources/views/modules/hot-post.blade.php]: thêm diffForHumans vào

File: storage/oauth-private.key [hunk 9]
```diff
@@ -1,51 +1,51 @@
 -----BEGIN RSA PRIVATE KEY-----
```
Comment (trantrongbinh) [inline storage/oauth-private.key]: đẩy file này lên làm gì?

File: storage/oauth-public.key [hunk 10]
```diff
@@ -1,14 +1,14 @@
 -----BEGIN PUBLIC KEY-----
-MIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAq68vCnWbXg3Z8VTWsWgl
```
Comment (trantrongbinh) [inline storage/oauth-public.key]: file này nữa.

File: app/Repositories/ArticleRepository.php [hunk 11]
```diff
@@ -325,4 +329,70 @@ public function getRelatedPosts($article)
 
         return $articles->get(['id', 'user_id', 'slug', 'page_image', 'published_at', 'title', 'category_id']);
     }
+
+    public function articleIndex()
+    {
+        $hotPosts  = $this->model->select('slug', 'title', 'user_id')
```
Comment (trantrongbinh) [inline app/Repositories/ArticleRepository.php]: cái này vất ra viewcomponent đi chứ để như thế này nhiều chỗ khác cũng gọi thì lại copy lại code này à.

File: app/Helpers.php (hunk 2 above)
Comment (trantrongbinh) [reply app/Helpers.php]: hoặc \Auth::id()

File: resources/views/article/show.blade.php (hunk 5 above)
Comment (0henri0) [reply resources/views/article/show.blade.php]: cái này null nó bị Erorr

File: resources/views/modules/hot-post.blade.php (hunk 8 above)
Comment (0henri0) [reply resources/views/modules/hot-post.blade.php]: $post->published_at == null sẽ bị erorr

File: storage/oauth-private.key (hunk 9 above)
Comment (0henri0) [reply storage/oauth-private.key]: hỏi t t biết hỏi ai

Comment (0henri0) [reply storage/oauth-private.key]: mày làm cái này mà

File: storage/oauth-public.key (hunk 10 above)
Comment (0henri0) [reply storage/oauth-public.key]: ko biết

File: app/Repositories/ArticleRepository.php (hunk 11 above)
Comment (0henri0) [reply app/Repositories/ArticleRepository.php]: ak cái này t định viết riêng nhưng mà thôi tí t xóa

File: resources/views/article/show.blade.php (hunk 5 above)
Comment (trantrongbinh) [reply resources/views/article/show.blade.php]: @0henri0  thế thì đừng cho nó, để default là timenow đi.

File: storage/oauth-private.key (hunk 9 above)
Comment (trantrongbinh) [reply storage/oauth-private.key]: checkout về như cũ đi.

Comment (trantrongbinh) [reply storage/oauth-private.key]: ko vứt vào .gitignore hộ phát

File: app/Article.php (hunk 1 above)
Comment (0henri0) [reply app/Article.php]: sao cái này nó tự động hiểu ak.

Comment (0henri0) [reply app/Article.php]: ví dụ chuyển hết model vào thư mục models thì nó có hiểu nữa ko

Comment (trantrongbinh) [reply app/Article.php]: oke. thường thì phải use.

File: app/Article.php [hunk 12]
```diff
@@ -192,4 +192,20 @@ public function setContentAttribute($value)
 
         $this->attributes['content'] = json_encode($data);
     }
+
+    public function upvoteUsers()
+    {
+        dd(category::class);
```
Comment (trantrongbinh) [inline app/Article.php]: vl dd()

File: app/Helpers.php (hunk 3 above)
Comment (trantrongbinh) [reply app/Helpers.php]: ý là: sum( count($post->upvoteUsers), count($post->downvoteUsers))
